datafusion/physical-expr-common/src/aggregate/mod.rs

```diff
@@ -573,8 +573,9 @@ impl AggregateExpr for AggregateFunctionExpr {
                     })
                     .collect::<Vec<_>>();
                 let mut name = self.name().to_string();
-                // TODO: Generalize order-by clause rewrite
-                if reverse_udf.name() == "ARRAY_AGG" {
+                // If the function is changed, we need to reverse order_by clause as well
+                // i.e. First(a order by b asc null first) -> Last(a order by b desc null last)
+                if self.fun().name() == reverse_udf.name() {
```

Comment:
I guess name checking is enough for now.

Introduce `supports_rewrite_order_by` for AggregateUDFImpl might add additional complexity without benefit.

Reply:
I think it would be good to eventually move this to a method in `https://github.com/apache/datafusion/pull/11611` though I agree this is good for now. Maybe we can file a ticket to track

Reply:
Thanks @alamb 
I come out a better idea of adding `reverse_name` method instead of `supports_rewrite_order_by ` which makes more sense to me. #11629